packages/recommend-react/src/useTrendingFacets.ts

```diff
@@ -23,33 +26,76 @@ export function useTrendingFacets<TObject>({
   });
   const { status, setStatus } = useStatus('loading');
 
-  useAlgoliaAgent({ recommendClient });
+  const { hasProvider, register } = useRecommendContext();
+  const { client, isContextClient } = useRecommendClient(recommendClient);
+
+  useAlgoliaAgent({ recommendClient: client });
 
   const transformItemsRef = useRef(userTransformItems);
   useEffect(() => {
     transformItemsRef.current = userTransformItems;
   }, [userTransformItems]);
 
   useEffect(() => {
-    setStatus('loading');
-    getTrendingFacets({
-      recommendClient,
-      transformItems: transformItemsRef.current,
+    const param: RecommendationsQuery = {
+      model: 'trending-facets',
       indexName,
-      maxRecommendations,
-      threshold,
+      // @ts-expect-error
```

Comment:
Can we provide an explanation after each `@ts-expect-error` to express why it's needed and to potentially better understand when it can be removed?

Reply:
Sarah and I didn't introduce the trending features, Sophie did. It's likely a wrong typing.

Reply:
I see, thanks for the clarification! I'll have a look at that type and maybe propose a change to fix the issue.